src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -937,26 +940,62 @@
                                                      "code_id"
                                                      (current-code-id-fn (get request-options "environment")))))})))
 
+(defn parse-project-compile-data
+  "Parse data required to compile a catalog inside a project. Data required includes
+   * Root path to project
+   * modulepath
+   * hiera config
+   * project_name"
+  [request-options
+   versioned-project
+   bolt-projects-dir]
+  (let [project-root (file-serving/get-project-root bolt-projects-dir versioned-project)
+        project-config (file-serving/read-bolt-project-config project-root)]
+    (assoc request-options "project_root" project-root
+                           "modulepath" (map #(str project-root "/" %) (file-serving/get-project-modulepath project-config))
+                           "hiera_config" (str project-root "/" (:hiera-config project-config))
```

Comment:
I really think we need to default to `hiera.yaml` here. I think think bolt defaults are going to be very difficult to keep track of, but one like this seems very hard to keep track of. Also, we should probably forbid absolute paths or trickery like `~,..` etc around this stuff.

Reply:
for now, i would almost reccomend we *only* accept a hiera.yaml at a project root.